lesson_18/homework_18_iterators.py

```diff
@@ -0,0 +1,46 @@
+# Ітератори:
+#
+# Реалізуйте ітератор для зворотного виведення елементів списку.
+# Напишіть ітератор, який повертає всі парні числа в діапазоні від 0 до N.
+
+class IteratorReverseList:
+    def __init__(self, initial_list: list):
+        self.current_position = 0
+        self.initial_list = initial_list
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.current_position <= 0 & self.current_position < self.initial_list.__len__():
+            self.current_position -= 1
+            return self.initial_list[self.current_position]
+        else:
+            return StopIteration
```

Comment:
when there are no elements left the re should be StopIteration exception
When I ran your code I got
```
Traceback (most recent call last):
  File "/Users/danilovets/Personal/Hillel/temp/AQA_Python_13082024/lesson_18/homework_18_iterators.py", line 48, in <module>
    print(next(iter1))
          ^^^^^^^^^^^
  File "/Users/danilovets/Personal/Hillel/temp/AQA_Python_13082024/lesson_18/homework_18_iterators.py", line 17, in __next__
    return self.initial_list[self.current_position]
           ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
IndexError: list index out of range
```

Reply:
expresion was updated. Need to be checked